src/main/java/com/couchbase/lite/router/Router.java

```diff
@@ -275,7 +275,14 @@ private Status openDB() {
         if (!db.exists()) {
             return new Status(Status.NOT_FOUND);
         }
-        if (!db.open()) {
+
+        boolean isOpen = false;
+        try {
+            isOpen = db.open();
```

Comment:
I remembered asking you about this before. Currently in the master branch, no exception thrown (only true / false) so no ways to know the cause of error. I believe we should throw an exception. iOS always has `out error` parameter to check what the error is.

While I'm working for SQLiteDatabase, Android always throws SQLiteDatabaseException or one of its subclass exception BUT they made them as a runtime exception. We could explore this option as well.